packages/ai-jsx/src/batteries/docs.tsx

```diff
@@ -739,21 +739,23 @@ const defaultLangchainChunkLimit = 4;
 async function searchVectorStore<ChunkMetadata extends Jsonifiable = Jsonifiable>(
   vectorStore: VectorStore,
   query: string,
-  params?: { limit?: number; score_threshold?: number }
+  params?: { limit?: number; score_threshold?: number; filter?: any }
 ): Promise<ScoredChunk<ChunkMetadata>[]> {
   const k = params?.limit ?? defaultLangchainChunkLimit;
-  const scoredLcDocs = await vectorStore.similaritySearchWithScore(query, k, _.omit(params, 'limit'));
-  return scoredLcDocs.map((lcDocAndScore) => {
-    const lcDoc = lcDocAndScore[0];
-    return {
-      score: lcDocAndScore[1],
-      chunk: {
-        // TODO: Wrap chunker to track document name in ChunkMetadata in a way we can pull back out here.
-        content: lcDoc.pageContent,
-        metadata: lcDoc.metadata as ChunkMetadata,
-      },
-    } as ScoredChunk<ChunkMetadata>;
-  });
+  const scoredLcDocs = await vectorStore.similaritySearchWithScore(query, k, params?.filter);
+  return scoredLcDocs
+    .map((lcDocAndScore) => {
+      const lcDoc = lcDocAndScore[0];
+      return {
+        score: lcDocAndScore[1],
+        chunk: {
+          // TODO: Wrap chunker to track document name in ChunkMetadata in a way we can pull back out here.
+          content: lcDoc.pageContent,
+          metadata: lcDoc.metadata as ChunkMetadata,
+        },
+      } as ScoredChunk<ChunkMetadata>;
+    })
+    .filter((chunk) => chunk.score >= (params?.score_threshold ?? Number.MIN_VALUE));
```

Comment:
@mdepinet here is my workaround fix for the type issue with newer versions of LC. The issue is that `VectorStore` doesn't take a single `filter` type as input anymore. It's dependent on the type of store that you use, e.g. Pinecone still takes a dictionary but `MemoryStore` takes a filter function as input and it throws an error if we provide an `object`.

Unfortunately, because of the way they do the typing, we don't get any TypeScript errors either way (so it can't easily be handled at compile time) and at runtime the types don't exist so we cannot decide what to pass in.
I decided to kinda put the burden on the user by saying that `params.filter` is what they provide and if it doesn't work then it's their fault.
This makes the interface a bit weird, but since nobody is currently using it, this should probably be fine. We can revisit later. WDYT?

Reply:
Seems like a reasonable approach to me. At the end of the day, we can only do so much to spare people from LC's garbage types if they want to use LC.